Add prefix directories in CRI-O mounts

FabienGhd opened the pull request:
Hi,

This PR addresses a TODO in <i>**policy.py**</i>, enhancing the handling of CRI-O mounts by adding support for a prefix directory:
> TODO(jaosorior): Add prefix-dir to path. This way we could call this from a container in kubernetes.

I basically updated the file to handle CRI-O mounts with or without a prefix directory for Kubernetes container calls.

Cheers,
Fabien

FabienGhd committed:
Add prefix directories in CRI-O mounts

Signed-off-by: Fabien Guihard <guihard.fa@gmail.com>

vmojzis commented:
Also please add "Sign off" to your commits ("git commit -s --amend").

vmojzis commented:
@JAORMX Hi, could you please take a look and confirm this will work with your workflow (as this was implemented based on your TODO)?

FabienGhd commented:
No worries about the delay! I had a small branch mix-up - sorry about that - but everything should be fine now.
Thanks!

vmojzis committed:
Clean up after adding prefix directories

- Rename the option to mount_prefix
- Apply the prefix to all container runtimes
- Add a test (although a dummy one)

Signed-off-by: Vit Mojzis <vmojzis@redhat.com>

vmojzis commented:
@JAORMX Hi, could you please take a look and confirm this will work with your workflow (as this was implemented based on your TODO).  I'd really appreciate an actual use case I could add to the tests.

JAORMX commented:
@vmojzis it's been a long time, I no longer remember what it was about tbh 😕

vmojzis commented:
Ok, no worries. In that case I'll leave this open for now since we don't have a clear use case (and close it in a few months unless someone comes up with up with one).
